src/inmanta/ast/statements/assign.py

```diff
@@ -145,15 +145,22 @@ def _resolve(self, requires: dict[object, object], resolver: Resolver, queue: Qu
 
     def execute_direct(self, requires: abc.Mapping[str, object]) -> object:
         qlist = []
-
         for i in range(len(self.items)):
             value = self.items[i]
-            qlist.append(value.execute_direct(requires))
+            item = value.execute_direct(requires)
```

Comment:
the execute_direct and as_constant did not do the proper flattening 

I also feel like we should merge the as_constant and direct_execute methods if at all possible, because both are so rarely used and similar it is hell to test them both